protocol/x/clob/memclob/memclob.go

```diff
@@ -935,13 +936,14 @@ func (m *MemClobPriceTimePriority) ReplayOperations(
 				continue
 			}
 
-			if m.operationsToPropose.IsOrderRemovalInOperationsQueue(statefulOrderPlacement.Order.OrderId) {
-				// If an order removal for this order is found in the ops queue, we have encountered an error. A
-				// stateful order should not be on the book if it hasn't been replayed yet which means it should not
-				// be possible for an order removal to precede a stateful order placement.
+			// Log an error if the order was already placed. We should not see duplicate Preexisting Stateful Orders
+			// and Order Removals (which, when replayed, can result in placements) should never precede a Preexisting
+			// Stateful Order operation.
+			if _, found := placedStatefulOrderIds[*orderId]; found {
 				m.clobKeeper.Logger(ctx).Error(
-					"ReplayOperations: Order removal found in operations queue for a stateful order which was not replayed yet.",
-					metrics.OrderId, statefulOrderPlacement.Order.OrderId,
+					"ReplayOperations: PreexistingStatefulOrder operation for order which was already placed",
+					metrics.OrderId, *orderId,
+					metrics.BlockHeight, ctx.BlockHeight(),
```

Comment:
This is technically `CheckTx` so it should be the block height of the next block right?

```suggestion
					metrics.BlockHeight, ctx.BlockHeight() + 1,
```